### PR TITLE
feat: bump Ruby version to 2.7

### PR DESCRIPTION
**Pull request by nekrich**
### Checklist

- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I see several green `ci/circleci` builds in the "All checks have passed" section of my PR ([connect CircleCI to GitHub](https://support.circleci.com/hc/en-us/articles/360008097173-Why-aren-t-pull-requests-triggering-jobs-on-my-organization-) if not)
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.
- [ ] I've added or updated relevant unit tests.

### Motivation and Context

Tried to [allow faraday v2](https://github.com/fastlane/fastlane/pull/22270), but it requires to bump Danger to 9.2, and Danger 9.2 requires Ruby 2.7
 

### Description

Dear reviewer, all hard work related to the Ruby version change is done in https://github.com/fastlane/fastlane/pull/22272/commits/b16d8423bb3df604f02fba10d11d07b83bbe1abf. And rubocop fixes are done in this commit: https://github.com/fastlane/fastlane/pull/22272/commits/265e2f44e0f8e6415b344afcb9512564502a56b8.

- Updated gemspec.
- Removed `fakefs` version limit, since the limit is no longer required.
- Removed unnecessary code from `ErbalT.render`.
- Updated CI: CircleCI and AppVeyor.
- Fixed linting problems with `rubocop -A` here -> https://github.com/fastlane/fastlane/pull/22272/commits/265e2f44e0f8e6415b344afcb9512564502a56b8.

### Testing Steps

In CI I trust 😅.

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # https://www.appveyor.com/docs/appveyor-yml/
 
 version: "{build}"
-
+image: Visual Studio 2022
```
Comment (nekrich) [inline appveyor.yml:4]: New image is required to use Ruby 2.7: [docs](https://www.appveyor.com/docs/windows-images-software/#ruby).

File: Gemfile [hunk 2]
```diff
@@ -10,16 +10,15 @@ gem "climate_control", "~> 0.2.0"
...
 gem "danger-junit", "~> 1.0"
 # A fake filesystem.
-# Version 1.9+ requires Ruby >=2.7, while fastlane uses a `required_ruby_version` of `>= 2.6`.
-gem "fakefs", "1.8"
+gem "fakefs", [">= 1.8", "< 3"]
 # for file uploads with Faraday
 gem "mime-types", ['>= 1.16', '< 4.0']
 # Fast XML parser and object marshaller.
-gem "ox", "2.13.2"
+gem "ox", "~> 2"
```
Comment (nekrich) [inline Gemfile:21]: Bundler was failing to build `ox, 2.13.2` as native extension on Windows.

File: fastlane/lib/fastlane/erb_template_helper.rb [hunk 3]
```diff
@@ -24,13 +24,7 @@ def initialize(hash, trim_mode = nil)
...
 
     def render(template)
-      # From Ruby 2.6, ERB.new takes keyword arguments and positional ones are deprecated
-      # https://bugs.ruby-lang.org/issues/14256
-      if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("2.6.0")
-        ERB.new(template, trim_mode: @trim_mode).result(binding)
-      else
-        ERB.new(template, nil, @trim_mode).result(binding)
-      end
+      ERB.new(template, trim_mode: @trim_mode).result(binding)
```
Comment (nekrich) [inline fastlane/lib/fastlane/erb_template_helper.rb:27]: Previous code was never used, since Ruby 2.6 requirement.

File: .circleci/config.yml [hunk 4]
```diff
@@ -286,8 +286,8 @@ workflows:
           ruby_opt: -W:deprecated
       - tests_ubuntu:
           name: 'Execute tests on Ubuntu'
-          image: 'fastlanetools/ci:0.3.0'
-          ruby_version: '2.6'
+          image: 'circleci/ruby:2.7.5-node'
```
Comment (nekrich) [inline .circleci/config.yml:289]: No need in custom docker container. `fastlanetools/ci` is locked on Ruby 2.6 and failing with v2.7.

File: appveyor.yml [hunk 5]
```diff
@@ -23,9 +23,12 @@ init:
...
 install:
-  - set PATH=C:\Ruby26-x64\bin;%PATH%
-  - FOR /F "usebackq delims==" %%A IN (`ruby .ci/compatible_gem_version`) DO gem update --system %%A
-  - FOR /F "usebackq delims==" %%A IN (`ruby .ci/bundler_version.rb`) DO gem install bundler -v %%A
+  - set PATH=C:\Ruby27-x64\bin;%PATH%
+  - ruby --version
+  - ruby .ci/compatible_gem_version
+  - FOR /F "usebackq delims==" %%A IN (`ruby .ci/compatible_gem_version`) DO gem update --no-document --system %%A
+  - ruby .ci/bundler_version.rb
+  - FOR /F "usebackq delims==" %%A IN (`ruby .ci/bundler_version.rb`) DO gem install bundler --no-document -v %%A
```
Comment (nekrich) [inline appveyor.yml:31]: - Add version info before trying to update. Was helpful while it was failing to use Ruby 2.7 on the older images.
- Add `--no-document` to speed it up a little bit.

**Comment by lacostej** [comment]
I prepared #21685 earlier. Anything we should recover from that work?

**Comment by nekrich** [comment]
> I prepared #21685 earlier. Anything we should recover from that work?

@lacostej 
I think it's better to use Visual Studio **2022** instead of 2019 on AppVeyor, in case they will skip some ruby versions on 2019.

And delete unnecessary code in fastlane/lib/fastlane/erb_template_helper.rb render. See 27-33 lines.

And do you have any plans to merge your PR in the nearest future?

**Comment by lacostej** [comment]
| And do you have any plans to merge your PR in the nearest future? Your PR is 10 months old 😅.

Well my understanding back then that we didn't want to drop ruby 2.6 until MacOS dropped ruby 2.6 or that the number of users using it was going down sufficiently. I made some heuristics to try to guess the %age of users. See the PR.

I was waiting for some feedback from other committers. Has anyone commented on this topic?

Also, I was hoping that @joshdholtz or someone else would release #fastlane/docker#8, but that's not blocking.

I'll look at your PR in the coming days.

**Comment by lacostej** [comment]
I just checked the percentage of tickets reported using ruby 2.6 and it is now below 4% (for tickets since March).

**Comment by nekrich** [comment]
Closing since there is a similar PR: https://github.com/fastlane/fastlane/pull/21685

**Pull request closed without merging**